src/internal/tab.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+  html,
+  css,
+  LitElement,
+  customElement,
+  property,
+  PropertyValues,
+  query,
+} from 'lit-element';
+
+/**
+ * A tab in a <playground-internal-tab-bar>.
+ *
+ * Slots:
+ * - default: Label or other contents of the tab.
+ *
+ * Parts:
+ * - button: Button with tab role.
+ */
+@customElement('playground-internal-tab')
+export class PlaygroundInternalTab extends LitElement {
+  static styles = css`
+    :host {
+      display: flex;
+    }
+
+    button {
+      flex: 1;
+      border: none;
+      color: inherit;
+      background: transparent;
+      display: flex;
+      align-items: center;
+      cursor: pointer;
+      position: relative;
+      outline: none;
+    }
+
+    button::before {
+      content: '';
+      position: absolute;
+      width: 100%;
+      height: 100%;
+      left: 0;
+      top: 0;
+      background: currentcolor;
+      opacity: 0;
+      transition: opacity 150ms;
+    }
+
+    button:focus::before,
+    button:hover::before {
+      opacity: 10%;
+    }
+
+    button:active::before {
+      opacity: 20%;
+    }
+  `;
+
+  /**
+   * Whether this tab is currently active.
+   */
+  @property({type: Boolean, reflect: true})
+  active = false;
+
+  @query('button')
+  private _button!: HTMLButtonElement;
+
+  /**
+   * The 0-indexed position of this tab within its <playground-internal-tab-bar>.
+   *
+   * Note this property is managed by the containing <playground-internal-tab-bar> and
+   * should not be set directly.
+   */
+  index = 0;
+
+  update(changes: PropertyValues) {
+    if (changes.has('active')) {
```

Comment:
What's the reasoning for doing this in update and not updated? is tabindex or aria-selected important to tab-bar at this point?

(Not a leading question, I genuinely don't know)